Data_tests callout (#5017)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/data-tests.md

* [Data test configurations](/reference/data-test-configs)
* [Test selection examples](/reference/node-selection/test-selection-examples)

<VersionBlock firstVersion="1.8">

:::important

> With the addition of unit tests in dbt v1.8, `tests` are now `data tests`. In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias but will be deprecated in the future in favor of `data_tests:`. See [New syntax](#new-syntax) for more information.

Author: ```suggestion
With the addition of unit tests in dbt v1.8, `tests` are now `data tests`. In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias but will be deprecated in the future in favor of `data_tests:`. See [New syntax](#new-`tests:`-syntax) for more information.
```

> With the addition of unit tests in dbt v1.8, `tests` are now `data tests`. In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias but will be deprecated in the future in favor of `data_tests:`. See [New syntax](#new-`tests:`-syntax) for more information.

Author: ```suggestion
With the addition of unit tests in dbt v1.8, what was previously called "tests" are now "data tests". In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias for data tests but will be deprecated in the future in favor of `data_tests:`. See [New syntax](#new-`tests:`-syntax) for more information.
```

> With the addition of unit tests in dbt v1.8, what was previously called "tests" are now "data tests". In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias for data tests but will be deprecated in the future in favor of `data_tests:`. See [New syntax](#new-tests-syntax) for more information.

Reviewer: sugg: slightly more concise and improved readability 

```suggestion
In dbt v1.8, what was previously known as "tests" are now called "data tests". The YAML key `tests:` is still supported as an alias for data tests but will be deprecated in the future in favor of `data_tests:`. Refer to [New syntax](#new-tests-syntax) for more information.
```

Author: I think it's important to call out that this is the result of a new test type. How about:
```suggestion
In dbt v1.8, what was previously known as "tests" are now called "data tests" with the addition of [unit tests](/docs/build/unit-tests). The YAML key `tests:` is still supported as an alias for data tests but will be deprecated in the future in favor of `data_tests:`. Refer to [New syntax](#new-tests-syntax) for more information.
```

Reviewer: love it!

> With the addition of unit tests in dbt v1.8, what was previously called "tests" are now "data tests". In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias for data tests but will be deprecated in the future in favor of `data_tests:`. See [New syntax](#new-`tests:`-syntax) for more information.

Author: ```suggestion
With the addition of unit tests in dbt v1.8, what was previously called "tests" are now "data tests". In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias for data tests but will be deprecated in the future in favor of `data_tests:`. See [New syntax](#new-tests-syntax) for more information.
```

* Test result tables are created in a schema suffixed or named `dbt_test__audit`, by default. It is possible to change this value by setting a `schema` config. (For more details on schema naming, see [using custom schemas](/docs/build/custom-schemas).)
- A test's results will always **replace** previous failures for the same test.

> <VersionBlock firstVersion="1.8">

Author: ```suggestion
<VersionBlock firstVersion="1.8" lastVersion="1.8">
```

> ## New `tests:` syntax

Reviewer 2: ```suggestion
## New `data_tests:` syntax
```

> With the addition of unit tests in dbt v1.8, `tests` are now `data tests`. In dbt v1.8, `tests:` is still supported in your YML configuration file as an alias but will be deprecated in the future in favor of `data_tests:`. 

Author: ```suggestion
Data tests were historically called "tests" in dbt as the only form of testing available. With the introduction of unit tests in v1.8, it was necessary to update our naming conventions and syntax. As of v1.8,  `tests:` is still supported in your YML configuration file as an alias but will be deprecated in the future in favor of `data_tests:`. 
```

As we progress towards this deprecation, the examples in our docs pages will be updated to reflect this new syntax, but we highly recommend you begin the migration process as soon as you upgrade to v1.8 to avoid interruptions or issues in the future.

> 

Reviewer: do you think it makes sense to move this section/content into the Upgrade to v1.8 guide? it can help make this page stay more evergreen sans version blocks

for "before and after" examples, we do have some prior art here (look at the second example) https://docs.getdbt.com/docs/dbt-cloud-apis/discovery-schema-environment#example-queries

Author: It will be there too but we need to call this out in the doc before we change all of the examples. Users may not look at that guide. I'll make this the last version as well to help

Reviewer: the content reads very much like "release note" material (snapshot in time, not evergreen)

or there plans to remove this section after some date? the syntax will eventually not be "new" anymore

Author: Yes, I version blocked it so it only shows in 1.8. Everything in the upgrade guide is elsewhere in the docs as well.

```yml

models:

> name: orders

Reviewer 2: the yml spacing looks a bit weird here

Reviewer 2: ![Screenshot 2024-02-29 at 7 08 53 PM](https://github.com/dbt-labs/docs.getdbt.com/assets/53586774/01db98d1-97cf-4b9b-a2fe-9cf2d6df03bf)
